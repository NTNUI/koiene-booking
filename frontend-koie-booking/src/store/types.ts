// rootState = generic type, defines the root state type
import AdminBooking, { AdminBookingDictionary } from '@/types/admin/AdminBooking';
import { AdminReportKoieDictionary } from '@/types/admin/AdminReports';

export interface RootState {
  version: string;
  koie: KoieState;
  booking: BookingState;
  auth: AuthState;
  avalanche: AvalancheState;
  adminBookings: AdminBookingsState;
}

export interface KoieState {
  currentLocale: string;
  isLoading: boolean;
  koieData: Koie;
  error: boolean;
  allKoier: [];
}

export interface Koie {
  location: {
    latitude: string;
    longitude: string;
    area: string;
    difficultyInfo_eng: string;
    difficultyInfo_nor: string;
    altitude: number;
    terrain_eng: string;
    terrain_nor: string;
    map_pdf: string;
    kartblad: string;
  };
  description: {
    yr_link_eng: string;
    yr_link_nor: string;
    description_eng: string;
    description_nor: string;
    directions_eng: string;
    directions_nor: string;
    parking_eng: string;
    parking_nor: string;
  };
  beds_available_in_booking_window: Array<BedsAvailable>;
  name: string;
  number_of_beds: number;
  booking_window: number;
  price_member: number;
  price_not_member: number;
}

export interface BedsAvailable {
  date: string;
  bedsAvailable: number;
}

export interface BookingState {
  isLoading: boolean;
  dateFrom: string;
  dateTo: string;
  beds: number;
  step: number;
  guests: Array<Guest>;
  numberOfMembers: number;
  numberOfNonMembers: number;
  edited: boolean;
  validForm: boolean;
  unavailableDates: [];
  availableBeds: number;
  noDatesAvailable: boolean;
  error: boolean;
  bookingData: BookingInfo;
  paymentConfirmed: boolean;
}

export interface ReportState {
  validForm: boolean;
  edited: boolean;
  isLoading: boolean;
  error: boolean;
  step: number;
  reportData: ReportData;
  bookingInfo: BookingInfo;
}

export interface ReportData {
  id: number;
  booking_uuid: string;
  firewood: number;
  chopped_up_wood: number;
  smoke_detector_is_working: boolean;
  gas_is_full: boolean;
  gas_burner_primus: number;
  axe: number;
  hammer: number;
  saw: number;
  saw_blade: number;
  saw_bench: number;
  spade: number;
  kerosene_lamp: number;
  detergent: number;
  dishware: number;
  cookware: number;
  cabin_book: number;
  candle_holders: number;
  fire_blanket: number;
  fire_extinguisher: number;
  other_faults: string;
  boat_status: number;
  canoe_status: number;
  life_jackets_status: number;
  feedback: string;
}

export interface AvalancheState {
  warningLevels: Array<Warning>;
  disableBooking: boolean;
  warningData: Array<WarningResult>;
  koierToCheck: Array<string>;
}

export interface Warning {
  day: string;
  level: number;
}

export interface WarningResult {
  dangerLevel: number;
  validTo: string;
}

export interface CreateBookingInfo {
  koie: string;
  from_date: string;
  to_date: string;
  guests: Array<Guest>;
  guests_member: number;
  guests_not_member: number;
}

export interface BookingInfo {
  id: number;
  uuid: string;
  user: number;
  koie: string;
  price: number;
  booking_transaction_id: string;
  from_date: string;
  to_date: string;
  guests_member: number;
  guests_not_member: number;
}

export interface AuthState {
  isAdmin: boolean;
  tokens: {
    access: string | null;
    refresh: string | null;
  };
}

export interface Guest {
  name: string;
  number: string;
  email: string;
  isMember: boolean;
}

export interface AdminBookingsState {
  cabinsWithBookings: AdminBookingDictionary;
  startDate: string;
}

export interface AdminReportsState {
  chosenCabin: string;
  expanded: boolean;
  cabins: AdminReportKoieDictionary;
}
