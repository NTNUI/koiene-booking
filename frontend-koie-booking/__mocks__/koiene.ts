import AdminBooking, { AdminBookingDictionary } from '../src/types/admin/AdminBooking';

export default function getKoieData(): AdminBookingDictionary {
  return koieData;
}

const koieData: AdminBookingDictionary = {
  ['flåkoia']: {
    name: 'Flåkoia',
    slug: 'flåkoia',
    numberOfBeds: 11,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 0,
      ['2020-09-16']: 0,
      ['2020-09-17']: 0,
      ['2020-09-18']: 11,
      ['2020-09-19']: 9,
      ['2020-09-20']: 0,
      ['2020-09-21']: 11,
      ['2020-09-22']: 11,
      ['2020-09-23']: 11,
      ['2020-09-24']: 11,
    },
  },
  ['mevasskoia']: {
    name: 'Mevasskoia',
    slug: 'mevasskoia',
    numberOfBeds: 5,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 5,
      ['2020-09-16']: 5,
      ['2020-09-17']: 5,
      ['2020-09-18']: 1,
      ['2020-09-19']: 0,
      ['2020-09-20']: 3,
      ['2020-09-21']: 5,
      ['2020-09-22']: 2,
      ['2020-09-23']: 5,
      ['2020-09-24']: 5,
    },
  },
  ['øvensenget']: {
    name: 'Øvensenget',
    slug: 'øvensenget',
    numberOfBeds: 8,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 8,
      ['2020-09-16']: 8,
      ['2020-09-17']: 8,
      ['2020-09-18']: 8,
      ['2020-09-19']: 8,
      ['2020-09-20']: 8,
      ['2020-09-21']: 8,
      ['2020-09-22']: 8,
      ['2020-09-23']: 8,
      ['2020-09-24']: 8,
    },
  },
  ['kåsen']: {
    name: 'Kåsen',
    slug: 'kåsen',
    numberOfBeds: 8,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 8,
      ['2020-09-16']: 8,
      ['2020-09-17']: 8,
      ['2020-09-18']: 8,
      ['2020-09-19']: 8,
      ['2020-09-20']: 8,
      ['2020-09-21']: 8,
      ['2020-09-22']: 8,
      ['2020-09-23']: 8,
      ['2020-09-24']: 8,
    },
  },
  ['stakkslettbua']: {
    name: 'Stakkslettbua',
    slug: 'stakkslettbua',
    numberOfBeds: 11,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 11,
      ['2020-09-16']: 11,
      ['2020-09-17']: 11,
      ['2020-09-18']: 11,
      ['2020-09-19']: 11,
      ['2020-09-20']: 11,
      ['2020-09-21']: 11,
      ['2020-09-22']: 11,
      ['2020-09-23']: 11,
      ['2020-09-24']: 11,
    },
  },
  ['holmsåkoia']: {
    name: 'Holmsåkoia',
    slug: 'holmsåkoia',
    numberOfBeds: 20,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 20,
      ['2020-09-16']: 20,
      ['2020-09-17']: 20,
      ['2020-09-18']: 20,
      ['2020-09-19']: 20,
      ['2020-09-20']: 20,
      ['2020-09-21']: 20,
      ['2020-09-22']: 20,
      ['2020-09-23']: 20,
      ['2020-09-24']: 20,
    },
  },
  ['sonvasskoia']: {
    name: 'Sonvasskoia',
    slug: 'sonvasskoia',
    numberOfBeds: 8,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 8,
      ['2020-09-16']: 8,
      ['2020-09-17']: 8,
      ['2020-09-18']: 8,
      ['2020-09-19']: 8,
      ['2020-09-20']: 8,
      ['2020-09-21']: 8,
      ['2020-09-22']: 8,
      ['2020-09-23']: 8,
      ['2020-09-24']: 8,
    },
  },
  ['nicokoia']: {
    name: 'Nicokoia',
    slug: 'nicokoia',
    numberOfBeds: 8,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 8,
      ['2020-09-16']: 8,
      ['2020-09-17']: 8,
      ['2020-09-18']: 8,
      ['2020-09-19']: 8,
      ['2020-09-20']: 8,
      ['2020-09-21']: 8,
      ['2020-09-22']: 8,
      ['2020-09-23']: 8,
      ['2020-09-24']: 8,
    },
  },
  ['telin']: {
    name: 'Telin',
    slug: 'telin',
    numberOfBeds: 9,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 9,
      ['2020-09-16']: 9,
      ['2020-09-17']: 9,
      ['2020-09-18']: 9,
      ['2020-09-19']: 9,
      ['2020-09-20']: 9,
      ['2020-09-21']: 9,
      ['2020-09-22']: 9,
      ['2020-09-23']: 9,
      ['2020-09-24']: 9,
    },
  },
  ['lyngli']: {
    name: 'Lyngli',
    slug: 'lyngli',
    numberOfBeds: 13,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 13,
      ['2020-09-16']: 13,
      ['2020-09-17']: 13,
      ['2020-09-18']: 13,
      ['2020-09-19']: 13,
      ['2020-09-20']: 13,
      ['2020-09-21']: 13,
      ['2020-09-22']: 13,
      ['2020-09-23']: 13,
      ['2020-09-24']: 13,
    },
  },
  ['kråklikåten']: {
    name: 'Kråklikåten',
    slug: 'kråklikåten',
    numberOfBeds: 4,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 4,
      ['2020-09-16']: 4,
      ['2020-09-17']: 4,
      ['2020-09-18']: 4,
      ['2020-09-19']: 4,
      ['2020-09-20']: 4,
      ['2020-09-21']: 4,
      ['2020-09-22']: 4,
      ['2020-09-23']: 4,
      ['2020-09-24']: 4,
    },
  },
  ['heinfjordstua']: {
    name: 'Heinfjordstua',
    slug: 'heinfjordstua',
    numberOfBeds: 25,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 25,
      ['2020-09-16']: 25,
      ['2020-09-17']: 25,
      ['2020-09-18']: 25,
      ['2020-09-19']: 25,
      ['2020-09-20']: 25,
      ['2020-09-21']: 25,
      ['2020-09-22']: 25,
      ['2020-09-23']: 25,
      ['2020-09-24']: 25,
    },
  },
  ['høgnabu']: {
    name: 'Høgnabu',
    slug: 'høgnabu',
    numberOfBeds: 6,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 6,
      ['2020-09-16']: 6,
      ['2020-09-17']: 6,
      ['2020-09-18']: 6,
      ['2020-09-19']: 6,
      ['2020-09-20']: 6,
      ['2020-09-21']: 6,
      ['2020-09-22']: 6,
      ['2020-09-23']: 6,
      ['2020-09-24']: 6,
    },
  },
  ['stabburet']: {
    name: 'Stabburet',
    slug: 'stabburet',
    numberOfBeds: 2,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 2,
      ['2020-09-16']: 2,
      ['2020-09-17']: 2,
      ['2020-09-18']: 2,
      ['2020-09-19']: 2,
      ['2020-09-20']: 2,
      ['2020-09-21']: 2,
      ['2020-09-22']: 2,
      ['2020-09-23']: 2,
      ['2020-09-24']: 2,
    },
  },
  ['vekvessætra']: {
    name: 'Vekvessætra',
    slug: 'vekvessætra',
    numberOfBeds: 20,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 20,
      ['2020-09-16']: 20,
      ['2020-09-17']: 20,
      ['2020-09-18']: 20,
      ['2020-09-19']: 20,
      ['2020-09-20']: 20,
      ['2020-09-21']: 20,
      ['2020-09-22']: 20,
      ['2020-09-23']: 20,
      ['2020-09-24']: 20,
    },
  },
  ['fosenkoia']: {
    name: 'Fosenkoia',
    slug: 'fosenkoia',
    numberOfBeds: 10,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 10,
      ['2020-09-16']: 10,
      ['2020-09-17']: 10,
      ['2020-09-18']: 10,
      ['2020-09-19']: 10,
      ['2020-09-20']: 10,
      ['2020-09-21']: 10,
      ['2020-09-22']: 10,
      ['2020-09-23']: 10,
      ['2020-09-24']: 10,
    },
  },
  ['tågåbu']: {
    name: 'Tågåbu',
    slug: 'tågåbu',
    numberOfBeds: 6,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 6,
      ['2020-09-16']: 6,
      ['2020-09-17']: 6,
      ['2020-09-18']: 6,
      ['2020-09-19']: 6,
      ['2020-09-20']: 6,
      ['2020-09-21']: 6,
      ['2020-09-22']: 6,
      ['2020-09-23']: 6,
      ['2020-09-24']: 6,
    },
  },
  ['lynhøgen']: {
    name: 'Lynhøgen',
    slug: 'lynhøgen',
    numberOfBeds: 4,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 4,
      ['2020-09-16']: 4,
      ['2020-09-17']: 4,
      ['2020-09-18']: 4,
      ['2020-09-19']: 4,
      ['2020-09-20']: 4,
      ['2020-09-21']: 4,
      ['2020-09-22']: 4,
      ['2020-09-23']: 4,
      ['2020-09-24']: 4,
    },
  },
  ['selbukåten']: {
    name: 'Selbukåten',
    slug: 'selbukåten',
    numberOfBeds: 2,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 2,
      ['2020-09-16']: 2,
      ['2020-09-17']: 2,
      ['2020-09-18']: 2,
      ['2020-09-19']: 2,
      ['2020-09-20']: 2,
      ['2020-09-21']: 2,
      ['2020-09-22']: 2,
      ['2020-09-23']: 2,
      ['2020-09-24']: 2,
    },
  },
  ['holvassgamma']: {
    name: 'Holvassgamma',
    slug: 'holvassgamma',
    numberOfBeds: 8,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 8,
      ['2020-09-16']: 8,
      ['2020-09-17']: 8,
      ['2020-09-18']: 8,
      ['2020-09-19']: 8,
      ['2020-09-20']: 8,
      ['2020-09-21']: 8,
      ['2020-09-22']: 8,
      ['2020-09-23']: 8,
      ['2020-09-24']: 8,
    },
  },
  ['kamtjønnkoia']: {
    name: 'Kamtjønnkoia',
    slug: 'kamtjønnkoia',
    numberOfBeds: 6,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 6,
      ['2020-09-16']: 6,
      ['2020-09-17']: 6,
      ['2020-09-18']: 6,
      ['2020-09-19']: 6,
      ['2020-09-20']: 6,
      ['2020-09-21']: 6,
      ['2020-09-22']: 6,
      ['2020-09-23']: 6,
      ['2020-09-24']: 6,
    },
  },
  ['rindalsløa']: {
    name: 'Rindalsløa',
    slug: 'rindalsløa',
    numberOfBeds: 4,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 4,
      ['2020-09-16']: 4,
      ['2020-09-17']: 4,
      ['2020-09-18']: 4,
      ['2020-09-19']: 4,
      ['2020-09-20']: 4,
      ['2020-09-21']: 4,
      ['2020-09-22']: 4,
      ['2020-09-23']: 4,
      ['2020-09-24']: 4,
    },
  },
  ['iglbu']: {
    name: 'Iglbu',
    slug: 'iglbu',
    numberOfBeds: 10,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 10,
      ['2020-09-16']: 10,
      ['2020-09-17']: 10,
      ['2020-09-18']: 10,
      ['2020-09-19']: 10,
      ['2020-09-20']: 10,
      ['2020-09-21']: 10,
      ['2020-09-22']: 10,
      ['2020-09-23']: 10,
      ['2020-09-24']: 10,
    },
  },
  ['mortenskåten']: {
    name: 'Mortenskåten',
    slug: 'mortenskåten',
    numberOfBeds: 2,
    bedsAvailableInDateRange: {
      ['2020-09-15']: 2,
      ['2020-09-16']: 2,
      ['2020-09-17']: 2,
      ['2020-09-18']: 2,
      ['2020-09-19']: 2,
      ['2020-09-20']: 2,
      ['2020-09-21']: 2,
      ['2020-09-22']: 2,
      ['2020-09-23']: 2,
      ['2020-09-24']: 2,
    },
  },
};
