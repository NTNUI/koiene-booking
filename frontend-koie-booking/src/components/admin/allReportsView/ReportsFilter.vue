<template>
  <v-col cols="11">
    <v-row justify="center">
      <h5 style="text-align: center;">Filtrer etter koie</h5>
    </v-row>
    <v-row justify="center">
      <v-btn text icon @click="setExpanded(!expanded)">
        <v-icon v-if="!expanded">mdi-arrow-down</v-icon>
        <v-icon v-else>mdi-arrow-up</v-icon>
      </v-btn>
    </v-row>
    <v-row v-show="expanded">
      <v-expand-transition>
        <v-row v-show="expanded" class="shrink d-flex flex-wrap justify-center">
          <template v-for="cabin in cabins">
            <v-chip
              v-if="chosenCabin !== cabin.slug"
              :key="cabin.slug"
              color="blue-grey darken-4"
              class="ma-1"
              @click="setChosenCabin(cabin.slug)"
              >{{ cabin.name }}</v-chip
            >
            <v-chip v-else :key="cabin.slug" color="primary" class="ma-1" @click="setChosenCabin('')">{{
              cabin.name
            }}</v-chip>
          </template>
        </v-row>
      </v-expand-transition>
    </v-row>
  </v-col>
</template>

<script lang="ts">
import Vue from 'vue';
import { sortObjectByKey } from '../../../utils/objects';

export default Vue.extend({
  name: 'ReportsFilter',
  computed: {
    cabins() {
      const unordered = this.$store.state.adminReports.cabins;
      const ordered = sortObjectByKey(unordered);
      return ordered;
    },
    chosenCabin() {
      return this.$store.state.adminReports.chosenCabin;
    },
    expanded() {
      return this.$store.state.adminReports.expanded;
    },
  },
  mounted() {
    this.mountCabins();
  },
  methods: {
    setChosenCabin(cabin: string) {
      this.$store.commit('adminReports/setChosenCabin', cabin);
      this.$store.dispatch('adminReports/MOUNT_REPORTS_FOR_CABIN', cabin);
    },
    setExpanded(expand: boolean) {
      this.$store.commit('adminReports/setExpanded', expand);
    },
    mountCabins() {
      this.$store.dispatch('adminReports/MOUNT_CABINS');
    },
  },
});
</script>

<style scoped></style>
