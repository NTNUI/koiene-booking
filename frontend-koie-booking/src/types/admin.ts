import { Component } from 'vue';

export interface AdminView {
  id: String; // Used to reference the buttons in SideBar
  title: String;
  icon: String;
  component: Component;
}
