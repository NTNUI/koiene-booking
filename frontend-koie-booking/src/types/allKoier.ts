export interface AllKoierData {
  koieTitle: string;
}
