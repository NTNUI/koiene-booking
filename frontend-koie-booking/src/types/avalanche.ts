import { TranslateResult } from 'vue-i18n';

export interface AvalancheData {
  result: string;
  levelFirstDay: string;
  levelSecondDay: string;
  levelThirdDay: string;
  firstDay: string;
  secondDay: string;
  thirdDay: string;
  fromDate: string;
  nextDate: string;
  lastDate: string;
  warningTypes: Array<TranslateResult>;
  colors: Array<string>;
}
