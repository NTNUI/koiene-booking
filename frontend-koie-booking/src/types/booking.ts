import { TranslateResult } from 'vue-i18n';

export interface BookingData {
  step: number;
  labels: Array<TranslateResult>;
}

export interface BookingFirstStepData {
  bedsTotal: number;
  bedsChosen: number;
  error: boolean;
}

export interface BookingSecondStepData {
  bedsChosen: number;
  bookingStep: number;
  guests: Array<Guest>;
  numberRules: object;
  nameRules: object;
  emailStrictRules: object;
  emailLooseRules: object;
  edited: boolean;
  validForm: boolean;
  memberPrice: number;
  nonMemberPrice: number;
}

export interface BookingThirdStepData {
  complete: boolean;
  stripeOptions: object;
  totalPrice: number;
}

export interface Guest {
  name: string;
  number: string;
  isMember: boolean;
  email?: string;
}
