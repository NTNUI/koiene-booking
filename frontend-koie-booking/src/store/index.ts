import Vue from 'vue';
import Vuex, { StoreOptions } from 'vuex';
import axios from 'axios';
import VueAxios from 'vue-axios';
import { RootState } from './types';
import { koie } from './modules/users/koie/index';
import { booking } from './modules/users/booking/index';
import { avalanche } from './modules/users/avalanche/index';
import { report } from './modules/users/report/index';
import { auth } from './modules/users/auth/index';
import { adminBookings } from '@/store/modules/admin/bookings';

Vue.use(Vuex, VueAxios, axios);

const store: StoreOptions<RootState> = {
  modules: {
    adminBookings,
    koie,
    booking,
    avalanche,
    report,
    auth
  }
};

export default new Vuex.Store<RootState>(store);

export { store as storeConfig };
