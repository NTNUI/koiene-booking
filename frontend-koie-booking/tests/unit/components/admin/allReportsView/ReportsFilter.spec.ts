import Vue, { VueConstructor } from 'vue';
import Vuetify, { Vuetify as VuetifyType } from 'vuetify';
import Vuex, { Store } from 'vuex';
import { storeConfig } from '@/store';
import { cloneDeep } from 'lodash';
import mockAxios from 'jest-mock-axios';

Vue.use(Vuetify);

// Utilities
import { Wrapper } from '@vue/test-utils';

// Components or views
import ReportsFilter from '@/components/admin/allReportsView/ReportsFilter.vue';

// Components or views
import { RootState } from '@/store/types';
import { MOCK_MOUNT_CABINS, MOCK_SET_CHOSEN_CABIN, MOCK_SET_EXPANDED } from '../../../__mocks__/koiene';
import { createWrapper } from '../../../utils';

describe('Component ReportsFilter.vue', () => {
  let wrapper: Wrapper<any>;
  let store: Store<RootState>;
  const setChosenCabin = jest.fn();
  const setExpanded = jest.fn();
  const mountCabins = jest.fn();

  beforeEach(() => {
    store = new Vuex.Store(cloneDeep(storeConfig));
    MOCK_MOUNT_CABINS(store);

    const wrapperOptions = {
      sync: false,
      store,
      methods: {
        setChosenCabin,
        setExpanded,
        mountCabins,
      },
    };

    wrapper = createWrapper(ReportsFilter, wrapperOptions);
  });

  afterEach(() => {
    mockAxios.reset();
  });

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Dispatches mountCabins', () => {
    expect(mountCabins).toHaveBeenCalled();
  });

  it('Calls setExpanded when button clicked with correct input', async () => {
    const button = wrapper.find('.v-btn');
    await button.trigger('click');
    expect(setExpanded).toHaveBeenCalledWith(true);
  });

  it('Should show filter chips when expanded is true', () => {
    MOCK_SET_EXPANDED(store, true);

    const newWrapperOptions = {
      sync: false,
      store,
    };

    wrapper = createWrapper(ReportsFilter, newWrapperOptions);
    expect(wrapper.findAll('.v-chip').isVisible()).toBe(true);
  });

  it('Sets chosen cabin on click with setChosenCabin', async () => {
    const cabinChip = wrapper.find('.v-chip:first-of-type');
    await cabinChip.trigger('click');
    expect(setChosenCabin).toHaveBeenCalledWith('flåkoia');
  });

  it('Should highlight chosen cabin chip', () => {
    const cabinSlug = 'flåkoia';
    MOCK_SET_CHOSEN_CABIN(store, cabinSlug);

    const newWrapperOptions = {
      sync: false,
      store,
    };

    wrapper = createWrapper(ReportsFilter, newWrapperOptions);

    const componentChosenCabin = wrapper.vm.chosenCabin;
    expect(componentChosenCabin).toEqual(cabinSlug);
  });

  it('Should hide filter chips on mount', () => {
    expect(wrapper.findAll('.v-chip').isVisible()).toBe(false);
  });
});
