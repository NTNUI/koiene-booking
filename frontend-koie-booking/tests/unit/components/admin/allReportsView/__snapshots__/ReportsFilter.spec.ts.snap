// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ReportsFilter.vue Matches snapshot 1`] = `
<div class="col col-11">
  <div class="row justify-center">
    <h5 style="text-align: center;">Filtrer etter koie</h5>
  </div>
  <div class="row justify-center"><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-arrow-down theme--light"></i></span></button></div>
  <div class="row" style="display: none;">
    <div class="row shrink d-flex flex-wrap justify-center" style="display: none;"><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Flåkoia</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Fosenkoia</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Heinfjordstua</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Holmsåkoia</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Holvassgamma</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Høgnabu</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Iglbu</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Kamtjønnkoia</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Kråklikåten</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Kåsen</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Lyngli</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Lynhøgen</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Mevasskoia</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Mortenskåten</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Nicokoia</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Rindalsløa</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Selbukåten</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Sonvasskoia</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Stabburet</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Stakkslettbua</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Telin</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Tågåbu</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Vekvessætra</span></span><span class="ma-1 v-chip v-chip--clickable theme--light v-size--default blue-grey darken-4"><span class="v-chip__content">Øvensenget</span></span></div>
  </div>
</div>
`;
